IntegrationTests/Services/AWSSTSIntegrationTests/STSAssumeRoleAWSCredentialIdentityResolverTests.swift

```diff
@@ -0,0 +1,130 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import XCTest
+import AWSSTS
+import AWSIAM
+import AWSSDKIdentity
+import ClientRuntime
+
+class STSAssumeRoleAWSCredentialIdentityResolverTests: XCTestCase {
+    private let region = "us-east-1"
+
+    // MARK: - Clients, policies, and variables
+
+    // STS client with only the STSAssumeRoleAWSCredentialIdentityResolver configured.
+    private var assumeRoleStsClient: STSClient!
+    private var assumeRoleStsConfig: STSClient.STSClientConfiguration!
+
+    // Used to create temporary role assumed by STS assume role credentials provider.
+    private var iamClient: IAMClient!
+    private let roleName = "aws-sts-integration-test-\(UUID().uuidString.split(separator: "-").first!.lowercased())"
+    private let roleSessionName = "aws-sts-integration-test-\(UUID().uuidString.split(separator: "-").first!.lowercased())"
+    private var roleArn: String!
+
+    // JSON assume role policy
+    private func assumeRolePolicy(acctID: String) -> String {
+        return """
+        {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Principal": {
+                        "AWS": "arn:aws:iam::\(acctID):root"
+                    },
+                    "Action": "sts:AssumeRole"
+                }
+            ]
+        }
+        """
+    }
```

Comment:
nit: consider extract this out to a helper function since assume role is also used in IMDS integration test

Reply:
[One in IMDS integ test](https://github.com/awslabs/aws-sdk-swift/blob/143d6938a2ec4997efd764daca2f57d70933a6cd/IntegrationTests/Services/AWSEC2IntegrationTests/IMDSAWSCredentialIdentityResolverTests.swift#L105-L110) is slightly different bc that one allows EC2 principal rather than account principal. I think it probably will require a function that needs multiple arguments with messy string concat logic just to handle the two cases.

But if more and more integration tests have this sort of duplicates, we can then consider refactoring integration test suite a bit and adding some utility class that can be used by any test for common uses like these.